tools/flight_recorder/components/utils.py

```diff
@@ -677,6 +689,27 @@ def get_version_detail(version: str) -> tuple[int, int]:
     return major, minor
 
 
+def add_stack_id_in_entries(
+    entries: dict[int, list[dict[str, Any]]],
+) -> tuple[dict[int, list[dict[str, Any]]], dict[str, int]]:
+    stack_id = 0
+    stack_id_trace_map = {}
+    for rank in entries:
+        for dump in entries[rank]:
+            if dump.get("frames", []):
+                frames = str(dump["frames"])
```

Comment:
can you remind me how the frames are stored in the on-disk file?  iirc they are already stored separately from the entries and there is some pointer/id that the entry uses to look up the frames in the frames table right?

if so, it would be nice to directly use this info so that we do not need to hash the large string of frames every time we check if one frames is in the map already